pkg/executor/test/indexmergereadtest/index_merge_reader_test.go

```
@@ -904,3 +904,29 @@ func TestIssues46005(t *testing.T) {
+		for i := rowID; i < rowID+500; i++ {
+			vals = append(vals, fmt.Sprintf("(1, 1, %d)", i))
+		}
+		tk.MustExec("insert into t(a,b,c) values " + strings.Join(vals, ","))
+	}
+
+	query := "select /*+ USE_INDEX_MERGE(t, idx1, idx2) */ * from t where a = 1 or b = 1 order by c limit 2000"
+	tk.MustHavePlan(query, "IndexMerge")
+	tk.MustNotHavePlan(query, "TopN")
+	// Issue #70910: the pushed-down limit must not be truncated to 1024 rows by
+	// the index merge union worker heap. The full 2000 rows must be returned.
+rows := tk.MustQuery(query).Rows()
```

golangci-lint (2.13.2)

[error] File is not properly formatted

(gofmt)